### PR TITLE
Admission webhook in Railgun

### LINKED ISSUE #1219: KIC 2.0 - admission webhook feature parity with KIC 1.x
Implement admission webhooks in KIC 2.0 to match KIC 1.x. Can probably reuse the existing implementation OOTB.

Can (maybe even should?) be done in a separate container/deployment.

### PR DESCRIPTION
**Pull request by mflendrich**
~Not done yet: testing. I'm planning to integration test the admission server before merging this PR.~

TL;DR: Enables the admission webhook in Railgun. 
Fixes #1219.

I came from an assumption that the admission webhook makes sense as a separate component than the controller manager (ideally a separate k8s deployment), but since the 1.x legacy is such that the admission webhook was an integral part of KIC, and to ease upgrades and limit breaking changes, it remains in the same binary/process, using the same flags. There are no intended breaking changes.

However, to maximize separation between the controller manager and the webhook server, I've separated it out to the greatest extent possible by running both separately from the top-level `Run()`.

The good:
- Wraps the TLS cert transformation logic in functions (as opposed to implemented inline in KIC 1.x main)
- Untangles the TLS cert path defaults handling a bit. The code no longer assumes that default cert path = no path set.
- Runs the admission webhook using the exact same config flags (one minor difference: the default value for cert/key paths is empty - unset - now, and filling in the defaults happens on another level)
- Makes it very straightforward to make the webhook server runnable as a standalone application (as long as it has access to a Kong with plugin schemas, but that may be runnable e.g. as a sidecar not serving traffic)
- Narrows down the `Storer` interface used by the validator to the necessary minimum (called it `SecretGetter`)

The not-sure-if-good:
- Relies on apiserver instead of the informer cache for the purpose of reading secrets.

The bad:
- To keep things simple, the admission webhook server creates its own kubeclient and kong client. This could be reused by the controller manager as well, but due to the potentially mutable nature of the kong client, and limited need for cross-wiring dependencies, I concluded that duplicating these won't be that much of a drag.

The ugly:
- The newly extracted functions in `package admission` are a logical copy of most of the inline logic in KIC 1.x `main`, but to limit potential regression blast in KIC 1.x I left the code in 1.x intact, and added a todo.

### CODE REVIEW

File: pkg/admission/server.go [hunk 1]
```diff
@@ -21,9 +22,77 @@ var (
...
+	switch {
+	case sc.CertPath != "" && sc.KeyPath != "" && sc.Cert == "" && sc.Key == "":
+		cert, key = []byte(sc.Cert), []byte(sc.Key)
+
+	case sc.CertPath == "" && sc.KeyPath == "" && sc.Cert != "" && sc.Key != "":
+		var err error
+		cert, key, err = readKeyPairFiles(sc.CertPath, sc.KeyPath)
+		if err != nil {
+			return nil, err
+		}
```
Comment (mflendrich) [inline pkg/admission/server.go:62]: add the default case returning error

**Review by rainest — APPROVED** [review]
LGTM. I don't expect much anything controversial in the forthcoming tests so not holding review on them.

Do note that the proposed indedepent admission webhook would still need to talk to the actual Kong instance it's going to approve configuration for--some checks (e.g. consumer collisions) rely on the state of configuration, so they wouldn't be able to use an empty instance that can just provide schemas.
